cns/NetworkContainerContract.go

```diff
@@ -338,26 +338,44 @@ type CreateNetworkContainerResponse struct {
 	Response Response
 }
 
-// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specifc network container.
+// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specific network container.
 type GetNetworkContainerStatusRequest struct {
 	NetworkContainerid string
 }
 
-// GetNetworkContainerStatusResponse specifies response of retriving a network container status.
+// GetNetworkContainerStatusResponse specifies response of retrieving a network container status.
 type GetNetworkContainerStatusResponse struct {
 	NetworkContainerid string
 	Version            string
 	AzureHostVersion   string
 	Response           Response
 }
 
-// GetNetworkContainerRequest specifies the details about the request to retrieve a specifc network container.
+// ContainerStatus is to save status of an existing container
+type ContainerStatus struct {
+	ID                            string
+	VMVersion                     string
+	HostVersion                   string
+	CreateNetworkContainerRequest CreateNetworkContainerRequest
+	VfpUpdateComplete             bool // True when VFP programming is completed for the NC
+}
+
+type GetOrRefreshNetworkContainersRequest struct {
+	CreateNetworkContainerRequests []CreateNetworkContainerRequest
+}
+
+type GetOrRefreshNetworkContainersResponse struct {
+	NetworkContainersDetails map[string]ContainerStatus // NetworkContainerID is the key.
+	Response                 Response
+}
```

Comment:
Some suggestions about the response struct. Usually, I'm of the opinion that IDs shouldn't be json keys, since they make the response less descriptive. I'd like to hear some thoughts from others, but I would recommend this struct be
```go
type GetNetworkContainersResponse struct {
    NetworkContainers []GetNetworkContainerResponse
    Response          Response
}
```

Reply:
Applied.